cache/cache.go

```diff
@@ -1,96 +1,72 @@
 package cache
 
-import (
-	"container/list"
-)
+import "container/list"
 
 // Node represents a node eligible for caching.
 type Node interface {
+	// GetKey returns a node's key
 	GetKey() []byte
+
+	// GetFullSize returns the number of bytes a node occupies in memory.
+	GetFullSize() int
 }
 
+// Type represents cache type. All types
+// are defined below.
+type Type int
+
+const (
+	LRU             Type = 0
+	LRU_node_limit  Type = 1
+	LRU_bytes_limit Type = 2
+)
```

Comment:
thoughts on switching this to iota syntax? https://yourbasic.org/golang/iota/

Reply:
Also are we using all three types?

Reply:
We are using the last 2 in iavl, and the first one is the abstract implementation of the last 2. 

I made it so that it can't be initialized outside of cache package. However, it still needs to have its own GetType() method

Reply:
Do we need the first one to be implemented? Perhaps we make a follow-up issue to delete it?

Reply:
Ohh the point is its needed, since we're doing things decorator style.

I guess I don't understand the type enum + decorator syntax combination. I don't think it needs to block the PR, but maybe we should make a follow-up issue about it

Reply:
The decorator pattern implies that LRU cache (the underlying implementation) can still be used on its own. We just don't use it in IAVL. Decorators are wrappers around the main abstraction to provide an additional layer of functionality. In our case, this functionality is limiting the cache.

We have three types implemented:
```
type Type int

const (
	LRU             Type = 0
	LRU_node_limit  Type = 1
	LRU_bytes_limit Type = 2
)
```

Only the last 2 are used in IAVL directly. However, this is composable by design - if we want the limit to be removed in the future - we just swap the cache for the regular LRU type

Let me know if this makes sense